### PR TITLE
Front end for Activity feed

### PR DESCRIPTION
**Pull request by aetheryna**
The components and data fetching :smile:

### CODE REVIEW

File: pages/api/post.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ const handler = nc()
     const newPost = new Post(postData)
 
     await newPost.createPost().catch((err) => {
+      console.log(err);
```
Comment (bterone) [inline pages/api/post.js]: The elusive console.log is here again 😂

File: src/components/PostModal.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import axios from "axios"
+import { useState } from "react"
+
+const PostModal = () => {
```
Comment (bterone) [inline src/components/PostModal.js]: Since we renamed PostModal to PostDropdown, this is no longer used 👍

File: pages/activity-feed/index.js [hunk 3]
```diff
@@ -15,4 +15,15 @@ const index = () => {
   )
 }
 
+export const getStaticProps = async() => {
+  const res = await fetch('http://localhost:3000/api/getAllPosts')
```
Comment (bterone) [inline pages/activity-feed/index.js:19]: Is it possible to use `await fetch('/api/getAllPosts')` instead? 🤔

File: pages/activity-feed/index.js [hunk 4]
```diff
@@ -15,4 +15,15 @@ const index = () => {
...
+  const postItems = await res.json()
```
Comment (bterone) [inline pages/activity-feed/index.js:20]: Do we need to use `await` here?

File: src/components/PostItem.js [hunk 5]
```diff
@@ -1,40 +1,52 @@
...
-            <Like />
-          </div>
-          <div className="post-item__svg">
-            <Favourite />
+        <div className="post-item__content mt-4 mb-2">
+          {post.image_url === "" ? 
+            ""
+            :
+            <div className="post-item__image mb-6">
+              <img className="mx-auto" src="https://picsum.photos/1600/900" alt="" />
```
Comment (bterone) [inline src/components/PostItem.js]: If we're not putting anything for the `alt`, then can we remove it? 😂

File: pages/api/post.js (hunk 1 above)
Comment (aetheryna) [reply pages/api/post.js]: 🐱

File: pages/activity-feed/index.js (hunk 3 above)
Comment (aetheryna) [reply pages/activity-feed/index.js:19]: Nope, NextJS needs an absolute URL for this one to work sadly 😞 

image

File: pages/api/post.js [hunk 6]
```diff
@@ -17,20 +17,36 @@ const handler = nc()
...
   .post(async(req, res) => {
     const data = req.body
+    
+    let imageData = {
+      image_url : '',
+    }
+
+    if (req.file) {
+      const filePath = req.file.path
+      const newFilePath = filePath.replace('public\\', '')
```
Comment (bterone) [inline pages/api/post.js]: Shall we just remove the `public\\` before saving it to the DB instead? On line 9 it seems like it evaluates code based on [their documentation ](https://www.npmjs.com/package/multer#storage)

```js
filename: (req, file, cb) => {
  cb(null, file.originalname.replace('public\\', ''))
},
```

That way we wouldn't need to add all these changes modifying the filePath 🤔

File: pages/activity-feed/index.js (hunk 4 above)
Comment (aetheryna) [reply pages/activity-feed/index.js:20]: Yup, it seems like the data cannot be fetched without a promise so we have to await it 🤷‍♂️ 

https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

image

File: src/components/PostItem.js (hunk 5 above)
Comment (aetheryna) [reply src/components/PostItem.js]: 🚀

**Review by bterone — APPROVED** [review]

**Pull request merged**